src/game.cpp

```diff
@@ -4872,6 +4872,10 @@ bool game::swap_critters( Creature &a, Creature &b )
 
     if( first.is_player() ) {
         update_map( *u_or_npc );
+
+        if( g->u.is_hauling() ) {
```

Comment:
I think `swap_critters()` should be rewritten to use `game::walk_move()` if `first.is_player()`, because then we get all the checks and effects that `walk_move()` performs.

Reply:
Fixed in #35583
Thanks!